core/string/string_builder.cpp

```diff
@@ -56,24 +56,32 @@ StringBuilder &StringBuilder::append(const char *p_cstring) {
 	return *this;
 }
 
+void StringBuilder::clear() {
+	string_length = 0;
+	strings.clear();
+	c_strings.clear();
+	appended_strings.clear();
+}
+
 String StringBuilder::as_string() const {
 	if (string_length == 0) {
 		return "";
 	}
 
-	char32_t *buffer = memnew_arr(char32_t, string_length);
+	String buffer;
+	buffer.resize(string_length + 1);
 
```

Comment:
You should retrieve the buffer pointer here and reuse it as every call to `buffer.ptrw()` and `buffer[]` goes through the COW checks. 

I.e.

```
char32_t * buffer_ptr = buffer.ptrw();
```